src/Compilers/Core/Portable/Symbols/ISymbolExtensions.cs

```diff
@@ -116,5 +117,20 @@ internal static bool IsInSource(this ISymbol symbol)
 
             return false;
         }
+
+        internal static bool IsTopLevelMainMethod([NotNullWhen(true)] this ISymbol? symbol)
```

Comment:
>IsTopLevelMainMethod [](start = 29, length = 20)

These are approximate checks which are fine for tests, but not for anything in product. They should be moved to location which is visible only to tests

Reply:
It looks like these helpers can be moved to ControlFlowGraphVerifier

---
In reply to: [529755288](https://github.com/dotnet/roslyn/pull/49534#discussion_r529755288) [](ancestors = 529755288)